### PR TITLE
refactor: add a repr helper function for the collections

### PR DESCRIPTION
**Pull request by TeamTamoad**
Implement an helper functions to for repr function of the collections including list, tuple, set and deque.

### CODE REVIEW

File: vm/src/utils.rs [hunk 1]
```diff
@@ -127,3 +127,40 @@ impl ToCString for PyStr {
         std::ffi::CString::new(self.as_ref()).map_err(|err| err.into_pyexception(vm))
     }
 }
+
+pub(crate) fn repr_helper<'a, I>(
+    class_name: Option<&str>,
+    brackets: [char; 2],
```
Comment (youknowone) [inline vm/src/utils.rs:133]: This variable is always used only for each char, but never as array. If we change this to `prefix: &str, suffix: &str`, we can handle more cases. For example, `deque` is now formatted again using maxlen. But if we just pass suffix string, we can pass right bracket and maxlen at same time.

**Review by youknowone — CHANGES_REQUESTED** [review]

**Comment by TeamTamoad** [comment]
Those changes are really good ideas. I'll do that.
Btw, do you have any idea why my last changes don't pass the test in time ?

**Comment by youknowone** [comment]
I have no idea yet. But becasue it failed again exceeding the time limit, it seems this changes made something.

**Comment by fanninpm** [comment]
> Btw, do you have any idea why my last changes don't pass the test in time ?

Try running the following command:

```text
cargo run --release --features ssl,jit -- -m test -v
```

**Comment by fanninpm** [comment]
[It was hanging in the snippets.](https://github.com/RustPython/RustPython/runs/4857239926) To diagnose,

1. Build RustPython in release mode (with the features normally used for testing): `cargo build --release --features ssl,jit`
2. Navigate to the `extra_tests` directory and install `pytest` into a virtual environment. (e.g., `cd extra_tests && pipenv install --python 3.9`)
3. Run pytest on the `extra_tests` directory. (e.g., `pipenv run pytest -v`)

**Comment by pwnorbitals** [comment]
how far are we from completion here @TeamTamoad ? :)

**Comment by TeamTamoad** [comment]
Sorry for an late update. I'm super busy with life and college right now.
The hanging problem is cause by this snippet in file `extra_tests/snippets/stdlib_collections.py`
```python
# Test that calling an evil __repr__ can't hang deque
class BadRepr:
    def __repr__(self):
        self.d.pop()
        return ''

b = BadRepr()
d = deque([1, b, 2])
b.d = d
repr(d)
```
It should be fixed now

**Review by youknowone — APPROVED** [review]
great! thank you for the long time effort!

**Pull request merged**